### PR TITLE
Task 644

### PR DESCRIPTION
**Pull request by ahmedelsayed-93**
https://github.com/gig-projects/org_quality/issues/644

### CODE REVIEW

File: functional_testing/Grid_API_Testing/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.ini
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/.gitignore]: why should we have .gitignore in our branch?

File: functional_testing/Grid_API_Testing/api_testing/grid_apis/grid_pyclient_base.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from g8os import resourcepool
+from api_testing.utiles.utiles import Utiles
+
+
+class GridPyclientBase(object):
+    def __init__(self):
+        self.config = Utiles().get_config_values()
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/grid_apis/grid_pyclient_base.py]: creat object from Utiles() and use it

File: functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            response = self.api_client.nodes.StartContainer(nodeid=nodeid, containername=containername, data={})
+        except HTTPError as e:
+            response = e.response
+        finally:
+            return response
+        
+    def post_containers_containerid_stop(self, nodeid, containername):
+        # work around
+        try:
+            response = self.api_client.nodes.StopContainer(nodeid=nodeid, containername=containername, data={})
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py]: they should modify their client to not use data={}!

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py [hunk 4]
```diff
@@ -311,17 +315,17 @@ def test007_post_storagepool_device(self):
         body = [device]
         response = self.storagepool_api.post_storagepools_storagepoolname_devices(self.nodeid, storagepool['name'], body)
         self.assertEqual(response.status_code, 204)
-        time.sleep(10)
+        time.sleep(30)
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py]: loop over it

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py [hunk 5]
```diff
@@ -337,11 +341,15 @@ def test008_delete_storagepool_device(self):
...
+        free_devices = self.pyclient.getFreeDisks()
+        if free_devices == []:
+            self.skipTest('no free disks on node {}'.format(self.nodeid))
+
+        device = random.choice(free_devices)
         body = [device]
         response = self.storagepool_api.post_storagepools_storagepoolname_devices(self.nodeid, storagepool['name'], body)
         self.assertEqual(response.status_code, 204)
-        time.sleep(10)
+        time.sleep(30)
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py]: loop over it

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py [hunk 6]
```diff
@@ -352,16 +360,16 @@ def test008_delete_storagepool_device(self):
         self.lg.info('Delete device (DV1), should succeed with 204')
         response = self.storagepool_api.delete_storagepools_storagepoolname_devices_deviceid(self.nodeid, storagepool['name'], deviceuuid[0])
         self.assertEqual(response.status_code, 204)
-        time.sleep(10)
+        time.sleep(20)
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py]: loop over it

File: functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py (hunk 3 above)
Comment (ahmedelsayed-93) [inline functional_testing/Grid_API_Testing/api_testing/grid_apis/pyclient/containers_apis.py]: I created issue in go-ramel repo

File: functional_testing/Grid_API_Testing/api_testing/grid_apis/grid_pyclient_base.py (hunk 2 above)
Comment (ahmedelsayed-93) [inline functional_testing/Grid_API_Testing/api_testing/grid_apis/grid_pyclient_base.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py (hunk 4 above)
Comment (ahmedelsayed-93) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py (hunk 5 above)
Comment (ahmedelsayed-93) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py (hunk 6 above)
Comment (ahmedelsayed-93) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test05_storagepools_apis.py]: done

**Pull request merged**